code/modules/clothing/suits/jobs.dm

```diff
@@ -213,40 +177,33 @@ obj/item/clothing/suit/apron/overalls/blue
 	desc = "A high-visibility vest used in work zones."
 	icon_state = "hazard"
 	item_state = "hazard"
-	var/icon_open = "hazard_open"
-	var/icon_closed = "hazard"
 	blood_overlay_type = "armor"
 	allowed = list (/obj/item/device/analyzer, /obj/item/device/flashlight, /obj/item/device/multitool, /obj/item/device/pipe_painter, /obj/item/device/radio, /obj/item/device/t_scanner, \
 	/obj/item/crowbar, /obj/item/screwdriver, /obj/item/weldingtool, /obj/item/wirecutters, /obj/item/wrench, /obj/item/tank/emergency_oxygen, \
 	/obj/item/clothing/mask/gas, /obj/item/taperoll/engineering)
 	body_parts_covered = UPPER_TORSO
+	var/opened
 
 /obj/item/clothing/suit/storage/hazardvest/verb/Toggle() //copied from storage toggle
 	set name = "Toggle Hazard Vest"
 	set category = "Object"
 	set src in usr
+
 	if(use_check_and_message(usr))
-		return FALSE
-	if(icon_state == icon_open)
-		item_state = icon_closed
-		icon_state = icon_closed
-		to_chat(usr, SPAN_NOTICE("You zip up the hazard vest."))
-	else if(icon_state == icon_closed)
-		item_state = icon_open
-		icon_state = icon_open
-		to_chat(usr, SPAN_NOTICE("You unzip the hazard vest."))
-	else
-		to_chat(usr, SPAN_NOTICE("You attempt to zip up your [src], before promptly realising how silly you are."))
-		return
+		return 0
+
+	opened = !opened
+	to_chat(usr, "You [opened ? "unzip" : "zip"] \the [src].")
```

Comment:
```suggestion
	to_chat(usr, SPAN_NOTICE("You [opened ? "unzip" : "zip"] \the [src]."))
```

Reply:
why isn't this a subtype of suit/storage/toggle actually?

Reply:
the difference is zipping and the sounds. can't figure out verbs for the life of me, though, and i don't plan on doing repathing of all the hazard vests in this PR.